src/arch/ia32e/address.rs

```diff
@@ -0,0 +1,84 @@
+pub struct VirtualAddress(u64);
+
+impl VirtualAddress {
+    /// Try to convert u64 into an ia32e virtual address.
+    ///
+    /// # Safety
+    ///
+    /// This function ensures the virtual address is correct.
+    pub fn try_new(addr: u64) -> Option<Self>
+    {
+        if addr < (1 << 48) {
```

Comment:
This rule is not sufficient to attest a valid address (every last bits after the 48th should be a 1). You may need to write a function to test this one.

Reply:
Done